lib/exq/adapters/queue/test.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Exq.Adapters.Queue.Test do
+  @moduledoc false
+
+  @behaviour Exq.Adapters.Queue
+
+  def enqueue(_pid, _queue, worker, args, _options) do
+    {:ok, apply(worker, :perform, args)}
```

Comment:
This is mostly a wish, not a requirement. We use `Exq.worker_job()` to get the job metadata like jid etc from within the worker. The test adapter would fail at that point. It would be great if we simulate that as well. We would have to do something similar to `Exq.Worker.Server#dispatch_work`.

Reply:
If we want to avoid a `if Mix.env() == :test` in worker_job or another sort of "adapter". Maybe we could bring **Exq**._worker_job_ from **Exq.Enqueuer.EnqueueApi**. The `use` would contain:

```elixir
  def worker_job(name \\ nil, pid \\ self()) do
    queue_adapter = Config.get(:queue_adapter)
    queue_adapter.worker_job(name, pid)
  end
```
In **Fake** queue:
```elixir
  def worker_job(name \\ nil, pid \\ self()) do
    %{jid: "123", queue: name}
  end
```

In **Async** queue the normal implementation:
```elixir
  def worker_job(name \\ nil, pid \\ self()) do
    metadata = Metadata.server_name(name)
    Metadata.lookup(metadata, pid)
  end
```

I understand it breaks the semantic of the queue which is about queue operations but it seems to be a small code change. I'm open to other ideas. Maybe you could explain more what you had in mind when you said:
> We would have to do something similar to Exq.Worker.Server#dispatch_work.

Reply:
I was thinking more on the lines of calling `Metadata.associate` from here.

``` elixir
job = construct_job(args)
:ok = Metadata.associate(metadata, self(), job)
result = apply(worker_worker, :perform, job.args)
```

But, I haven't looked through the details, like whether the metadata server will be up during test etc

Reply:
I had a look to this code but not sure to know where you're heading. Let me know you're idea and I'll be happy to help if I can